packages/psss/src/api/queries/useSingleWeeklyReport.js

```diff
@@ -0,0 +1,130 @@
+  alerts.reduce(
+    (list, syndrome) => (verifiedStatuses.includes(syndrome.id) ? list : [...list, syndrome.id]),
+    [],
+  );
+
+const getEmptyTableData = () => [
+  { id: 'afr', title: 'Acute Fever and Rash (AFR)', percentageChange: 0, totalCases: 0 },
+  { id: 'dia', title: 'Diarrhoea (DIA)', percentageChange: 0, totalCases: 0 },
+  { id: 'ili', title: 'Influenza-like Illness (ILI)', percentageChange: 0, totalCases: 0 },
+  { id: 'pf', title: 'Prolonged Fever (PF)', percentageChange: 0, totalCases: 0 },
+  { id: 'dli', title: 'Dengue-like Illness (DLI)', percentageChange: 0, totalCases: 0 },
+];
+
+const toCommaList = values =>
+  values
+    .join(', ')
+    .toUpperCase()
+    .replace(/,(?!.*,)/gim, ' and');
+
+const getTableData = data => [
+  {
+    id: 'afr',
+    title: 'Acute Fever and Rash (AFR)',
+    percentageChange: 0,
+    totalCases: data.AFR,
+    isAlert: data[`AFR Threshold Crossed`],
+  },
+  {
+    id: 'dia',
+    title: 'Diarrhoea (DIA)',
+    percentageChange: 0,
+    totalCases: data.DIA,
+    isAlert: data[`DIA Threshold Crossed`],
+  },
+  {
+    id: 'ili',
+    title: 'Influenza-like Illness (ILI)',
+    percentageChange: 0,
+    totalCases: data.ILI,
+    isAlert: data[`ILI Threshold Crossed`],
+  },
+  {
+    id: 'pf',
+    title: 'Prolonged Fever (PF)',
+    percentageChange: 0,
+    totalCases: data.PF,
+    isAlert: data[`PF Threshold Crossed`],
+  },
+  {
+    id: 'dli',
+    title: 'Dengue-like Illness (DLI)',
+    percentageChange: 0,
+    totalCases: data.DLI,
+    isAlert: data[`DLI Threshold Crossed`],
+  },
+];
+
+export const useSingleWeeklyReport = (orgUnit, period, verifiedStatuses, pageQueryKey) => {
+  const query = useData(
+    `weeklyReport/${orgUnit}`,
+    {
+      params: { startWeek: period, endWeek: period },
+    },
+    {
+      initialData: () => {
+        const cachedQuery = queryCache.getQueryData([`weeklyReport/${orgUnit}`, pageQueryKey]);
+        const results = cachedQuery?.data?.results;
+
+        if (!results) {
+          return undefined;
+        }
+
+        if (results.length === 0) {
+          return cachedQuery;
+        }
+
+        const record = results.filter(row => row.period === period);
+        return record.length > 0
+          ? { ...cachedQuery, data: { ...cachedQuery.data, results: record } }
+          : undefined;
+      },
+    },
+  );
+
+  if (query.isLoading || query.data.length === 0) {
+    return {
+      ...query,
+      data: {
+        Sites: 0,
+        'Sites Reported': 0,
+      },
+      syndromes: getEmptyTableData(),
+      alerts: [],
+      unVerifiedAlerts: [],
+      unVerifiedList: null,
+    };
+  }
+
+  const data = query.data[0];
+  const alerts = getAlerts(data);
+  const unVerifiedAlerts = getUnVerifiedAlerts(alerts, verifiedStatuses);
+  const syndromes = getTableData(data);
+
+  return {
+    ...query,
+    data,
+    syndromes,
+    alerts,
+    unVerifiedAlerts,
+    unVerifiedList: toCommaList(unVerifiedAlerts),
```

Comment:
It probably makes more sense to just return a list of alert strings in this function, and stringify them closer to the presentational layer (in the component)